src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java

```diff
@@ -61,68 +59,64 @@
-                                caseFieldDefaultValue,
-                                caseField.getCaseFieldId(),
-                                errorList);
-                            return true;
-                        } else {
-                            return false;
-                        }
-                    }
-                ).collect(Collectors.toList()))
-        );
+        caseDefinitionRepository.getCaseType(caseTypeId).getEvents().stream()
+            .filter(event -> event.getId().equals(content.getEventId()))
+            .forEach(caseEventDefinition -> caseEventDefinition.getCaseFields()
+                .stream()
+                .forEach(eventFieldDefinition -> eventFieldDefinition.getCaseEventFieldComplexDefinitions().stream()
+                    .filter(cefcDefinition -> isOrgPolicyCaseAssignedRole(cefcDefinition.getReference()))
+                    .forEach(cefcDefinition -> {
+                        //get extract the default value  from the content for the current eventFieldDefinition
+                        String reference = cefcDefinition.getReference();
+                        validateContent(content, eventFieldDefinition.getCaseFieldId(),
+                            reference,
+                            cefcDefinition.getDefaultValue(),
+                            errorList);
+                    })));
         if (errorList.size() != 0) {
             throw new ValidationException("Roles validation error: " + String.join(", ", errorList));
         }
     }
 
-    private boolean isOrganisationPolicyLogicInTheContent(final CaseDataContent content) {
-        final JsonNode existingData = new ObjectMapper().convertValue(content.getData(), JsonNode.class);
-        final List<JsonNode> jsonNode = existingData.findParents(ORGANISATION_POLICY_ROLE);
-        final Optional<JsonNode> node = jsonNode.stream().findFirst();
-        return  node.isPresent();
+    private boolean isOrgPolicyCaseAssignedRole(String reference) {
+        String[] referenceArray = reference.split(Pattern.quote("."));
+        return ORGANISATION_POLICY_ROLE.equals(referenceArray[referenceArray.length - 1]);
     }
 
-    private Optional<String> getDefaultValueFromContentByCaseFieldID(final CaseDataContent content, final String caseFiledID) {
+    private void validateContent(final CaseDataContent content,
+                                 final String caseFiledID,
+                                 final String reference,
+                                 final String defaultValue,
+                                 final List<String> errorList) {
         final JsonNode existingData = new ObjectMapper().convertValue(content.getData(), JsonNode.class);
-        final Optional<JsonNode> caseFieldNode = Optional.ofNullable(existingData.get(caseFiledID));
-
+        Optional<JsonNode> caseFieldNode = Optional.of(existingData.findPath(caseFiledID));
         if (caseFieldNode.isPresent()) {
```

Comment:
Overcomplicating by wrapping in an `Optional` rather than just checking not `null`?

Reply:
removed optional